src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1093,6 +1094,19 @@ public static int canEntitySpawn(Mob entity, LevelAccessor world, double x, doub
         return res == Result.DEFAULT ? 0 : res == Result.DENY ? -1 : 1;
     }
 
+    public static boolean hasNoElements(Ingredient ingredient)
+    {
+        ItemStack[] items = ingredient.getItems();
+        if (items.length == 0) return true;
+        if (items.length == 1 && !ForgeConfig.SERVER.treatEmptyTagsAsAir.get())
+        {
+            //If we potentially added a barrier due to the ingredient being an empty tag, try and check if it is the stack we added
+            ItemStack item = items[0];
+            return item.getItem() == Items.BARRIER && item.getHoverName() instanceof TextComponent hoverName && hoverName.getText().startsWith("Empty Tag: ");
```

Comment:
Would it be worth just adding a custom NBT tag to the forge item for empty tags? Something like `forge_empty_tag: "forge:ingots/iron"` and checking for that custom tag? This feels super hacky as its comparing displayable text. The tag would also be useful for things like recipe viewers to easily detect it without their own display name check hacks.

Reply:
The reason we used display text is because if for some reason someone ever got this item in a GUI. Such as NEI/RecipeBook it would tell them what tag it wants. Without having to lookup the NBT tags.

Reply:
Yeah, was not suggesting the display text be removed. Suggesting the extra tag in addition to the display text

Reply:
It feels wrong yes, but its no more wrong then this ugly ugly hack. Which I hope one day never has to happen because Mojang will of fixed their issues.